packages/rollup-plugin/src/index.ts

```diff
@@ -1,14 +1,51 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 import {
-  codecovUnpluginFactory,
+  type UnpluginOptions,
+  createRollupPlugin,
+  type RollupPlugin,
+} from "unplugin";
+import {
+  normalizeOptions,
+  red,
   type Options,
+  checkNodeVersion,
 } from "@codecov/bundler-plugin-core";
 
 import { rollupBundleAnalysisPlugin } from "./rollup-bundle-analysis/rollupBundleAnalysisPlugin";
 
-const codecovUnplugin = codecovUnpluginFactory({
-  bundleAnalysisUploadPlugin: rollupBundleAnalysisPlugin,
-});
+const codecovRollupPluginFactory = createRollupPlugin<Options, true>(
+  (userOptions, unpluginMetaContext) => {
+    const plugins: UnpluginOptions[] = [];
+
+    const normalizedOptions = normalizeOptions(userOptions);
+
+    if (!normalizedOptions.success) {
+      for (const error of normalizedOptions.errors) {
+        red(error);
+      }
+      return [];
+    }
+
+    if (checkNodeVersion(unpluginMetaContext)) {
+      return [];
```

Comment:
It's safer to check node version before running `normalizeOptions`.

We can also move `const plugins: UnpluginOptions[] = [];` down below the early returns considering we are just returning `[]` here.